Vault/ClientFactory.cs

```diff
@@ -24,17 +25,32 @@ public class ClientFactory
     
     public ClientFactory(
         string uriString = DefaultUriString,
-        string userKey = InitialAdminUserKey)
+        string userKey = InitialAdminUserKey,
```

Comment:
Why not letting the customer provide their own `HttpClient` and be done with it and all future requests?